cranelift/codegen/src/legalizer/mod.rs

```diff
@@ -779,7 +779,7 @@ fn narrow_icmp_imm(
         .iconst(ty_half, imm & ((1u128 << ty_half.bits()) - 1) as i64);
     let imm_high = pos
         .ins()
-        .iconst(ty_half, imm.wrapping_shr(ty_half.bits().into()));
+        .iconst(ty_half, imm.checked_shr(ty_half.bits().into()).unwrap_or(0));
```

Comment:
Do we need to mask this as well? `imm` is an `i64`, and so a right-shift is an arithmetic right-shift that replicates the MSB on the left; a quick test shows that `-1i64 >> 32` is still `-1i64` (i.e., all ones).

Reply:
Fixed